include/deal.II/grid/cell_id.h

```diff
@@ -82,7 +82,7 @@ class CellId
    * and the number of children of a cell in the current space dimension (i.e.,
    * GeometryInfo<dim>::max_children_per_cell).
    */
-  CellId(const unsigned int               coarse_cell_id,
+  CellId(const unsigned long int          coarse_cell_id,
```

Comment:
Rather than picking a type here, let's give it a name in `types.h`: `types::coarse_cell_id`. This way, the type is also self-documenting, and easy to change. Indeed, I think you should make the type 32bit by default, and only 64bit if we have `DEAL_II_WITH_64BIT_INDICES`: It doesn't make sense to have more than 4B coarse mesh cells if you don't also allow more than 4B DoFs.